File: frontend/src/types/risk.ts
import type { Confidence, ProviderId, Severity, UserRole } from "./common";

export type AlertType = "LIQUIDITY_PRESSURE" | "UNUSUAL_ACTIVITY" | "DATA_QUALITY" | "COMBINED_OPERATIONAL_REVIEW";
export type ActionCategory = "VERIFY_DEMAND" | "CONTACT_PROVIDER_OPERATIONS" | "COMPARE_CONTEXT" | "CONTINUE_MONITORING" | "ESCALATE_REVIEW";

export interface RiskAssessment {
  riskAssessmentId: string;
  agentId: string;
  providerId: ProviderId;
  riskLevel: Severity;
  alertType: AlertType;
  priority: Severity;
  confidence: Confidence;
  reasons: string[];
  allowAIAdvisory: boolean;
  requiredHumanRole: UserRole;
  allowedActionCategories: ActionCategory[];
  prohibitedActionCategories: string[];
  ruleVersion: string;
}

File: backend/app/schemas/risk.py
from __future__ import annotations

from pydantic import Field

from app.schemas.common import AgentLensSchema
from app.schemas.enums import AlertType, Severity, UserRole


class RiskAssessment(AgentLensSchema):
    severity: Severity
    alert_type: AlertType
    priority: int = Field(ge=1, le=4)
    reasons: list[str]
    confidence: float = Field(ge=0.0, le=1.0)
    allow_ai_advisory: bool
    required_human_role: UserRole
    allowed_actions: list[str]
    prohibited_actions: list[str]
    rule_version: str
    advisory_only: bool = True
